### PR TITLE
[FLINK-31269] Split hive connector to each module of each version

### PR DESCRIPTION
**Pull request by FangYongs**
This pr aims to split hive connector to each module of each version. 

Besides connector, there are different versions of hive catalog (hive 2 and 3). Hive3 metastore can be connected through the client of hive2.3.9. We do not need to create independent modules for catalog. If necessary, we can also divide the hive catalog into sub-modules in the future.

### CODE REVIEW

**Comment by JingsongLi** [comment]
I'm mainly worried about whether Hive-catalog can work properly. Have you tested all Hive versions?

**Comment by FangYongs** [comment]
> 

@JingsongLi I setup `metastore` in `Hive3.1.2` and use `flink-table-store-hive-catalog.jar` which is compiled by `Hive-2.3.9` in flink to execute `DDL` such as `CREATE` and `SHOW TABLES`, it works. Of course, splitting the catalog into different versions is also an option, what do you think?

**Comment by JingsongLi** [comment]
> 

Have you tried to setup metastore in hive-2.1-cdh-6.3, and use flink-table-store-hive-catalog.jar in Flink side?

**Comment by FangYongs** [comment]
Thanks @tsreaper  DONE

**Comment by FangYongs** [comment]
> > 
> 
> Have you tried to setup metastore in hive-2.1-cdh-6.3, and use flink-table-store-hive-catalog.jar in Flink side?

Hi @JingsongLi  I also test for hive-2.1.1, it works

File: docs/content/docs/engines/hive.md [hunk 1]
```diff
@@ -137,7 +131,7 @@ SELECT * FROM test_table;
 Run the following Hive SQL in Hive CLI to access the created table.
 
 ```sql
--- Assume that flink-table-store-hive-connector-{{< version >}}.jar is already in auxlib directory.
+-- Assume that flink-table-store-hive-connector-2.3-{{< version >}}.jar is already in auxlib directory.
```
Comment (tsreaper) [inline docs/content/docs/engines/hive.md:134]: `2.3` -> `<hive-version>`. There are other appearance of `flink-table-store-hive-connector-{{< version >}}.jar` in the document. Please replace them.

File: flink-table-store-hive/flink-table-store-hive-connector-2.1-cdh-6.3/pom.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    <execution>
+                        <id>shade-flink</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <artifactSet>
+                                <includes combine.children="append">
+                                    <include>org.apache.flink:flink-table-store-hive-connector</include>
```
Comment (tsreaper) [inline flink-table-store-hive/flink-table-store-hive-connector-2.1-cdh-6.3/pom.xml:63]: Rename the old module to `flink-table-store-hive-connector-common`.

**Comment by FangYongs** [comment]
Thanks @tsreaper Updated

**Pull request merged**